zkevm-circuits/src/keccak_circuit/keccak_packed_multi.rs

```diff
@@ -12,12 +12,43 @@ use std::{env::var, vec};
 
 const MAX_DEGREE: usize = 9;
 
-pub(crate) fn get_num_rows_per_round() -> usize {
+/// Obtain the rows required for 1 iteration of f-box's inner round
+/// function (consisting of 5 phases) within Keccak circuit
+pub fn get_num_rows_per_round() -> usize {
     var("KECCAK_ROWS")
```

Comment:
from the test it seems that the max number `KECCAK_ROWS` supports is 32. can we please add a check for this? and also add it to the document?

ALso for `KECCAK_ROWS=32` we are looking at `47` advice columns. And with `k=19`, we can support  roughly `655` rounds. (while `MAX_KECCAK_ROWS = 15000` at the moment) Is this sufficient @lispc

Reply:
what's MAX_KECCAK_ROWS?

Reply:
https://github.com/scroll-tech/zkevm-circuits/blob/2e422878e0d78f769e08f0b1ad1275ee039362d5/integration-tests/src/integration_test_circuits.rs#L67

Reply:
These parameter don't need to be same for super circuits and later circuits.   So for compression/aggregation circuit you can choose any parameter suitable.

Reply:
A lower bound check is added. From latest changes on test serial_keccak_circuit_unusable_rows it seems this variable is no longer required to be upper-bounded. Rows greater than 32 also works fine. 

Should a theoretical upper bound be added?

Reply:
maybe not.  Sometimes we may use KECCAK_ROWS = 50 to reduce width radically.  KECCAK_ROWS > 50 will waste cells but brings no benefits.  All values in range [9, 50] will be reasonable choice.